Week3/app/src/main/java/com/kulloveth/businesscard/MainActivity.kt

```diff
@@ -0,0 +1,221 @@
+            binding.emailIv.imageTintList = ContextCompat.getColorStateList(this, styeTHeme)
+            supportActionBar?.setBackgroundDrawable(ColorDrawable(resources.getColor(styeTHeme)))
+            binding.tipsBtn.backgroundTintList = ContextCompat.getColorStateList(this, styeTHeme)
+        }
+        binding.languageTipsTv.text = currentTips
+        binding.languageIv.setImageResource(imageRes)
+    }
+
+
+    /**
+     * create the toolbar [menu] items
+     * by referencing the resource file
+     * */
+    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
+
+        //this adds item to the toolbar if present
+        menuInflater.inflate(R.menu.main_menu, menu)
+        return super.onCreateOptionsMenu(menu)
+    }
+
+    /**
+     * create actions for the menu item
+     * by referencing its it through [item]
+     */
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        when (item.itemId) {
+            R.id.filter -> optionDialog()
+            R.id.about -> aboutDialog()
+            R.id.share -> shareTips()
+        }
+        return super.onOptionsItemSelected(item)
+    }
+
+    //creates an intent share feature
+    private fun shareTips() {
+        startActivity(
+            Intent.createChooser(
+                ShareCompat.IntentBuilder.from(this)
+                    .setType("text/plain")
+                    .setText(currentTips)
+                    .intent, getString(toolbarTitle) + " tips"
+            )
+        )
+    }
+
+    //create dialog to show app information
+    private fun aboutDialog() {
+        val alertdialog = AlertDialog.Builder(this)
+        alertdialog.apply {
+            setTitle(getString(R.string.app_title))
+            setMessage(getString(R.string.developer))
+            setPositiveButton("OK") { dialog, _ -> dialog.dismiss() }
+            show()
+        }
+
+
+    }
+
+    /**
+     * saves the data state using [bundle]
+     * to ensure its not recreated on orientation change
+     * */
+    override fun onSaveInstanceState(bundle: Bundle) {
+        super.onSaveInstanceState(bundle)
+        bundle.putString(TIPS_KEY, currentTips)
+        bundle.putInt(IMAGE_KEY, imageRes)
+        bundle.putInt(Title_KEY, toolbarTitle)
+        bundle.putInt(TOOLBAR_KEY, styeTHeme)
+        bundle.putString(LANGUAGE_KEY, language)
+        bundle.putStringArrayList(LIST_KEY, tipsList.toList() as ArrayList<String>)
+    }
+
+    /**
+     * creates an Alertdialog with different
+     * language otion to select from
+     * */
+    private fun optionDialog() {
+        val alertDialog = AlertDialog.Builder(this)
+        alertDialog.setTitle(getString(R.string.select_language))
+        val items = arrayOf(
+            resources.getString(R.string.kotlin_language),
+            resources.getString(R.string.python_language),
+            resources.getString(R.string.java_language)
+        )
+
+
+        alertDialog.setSingleChoiceItems(items, -1)
```

Comment:
You've gone above and beyond here by displaying a menu with a choice and maintaining the state of that menu throughout an orientation change, excellent job.